src/mpi/coll/bcast.c

```diff
@@ -1095,28 +1079,43 @@ static int MPIR_SMP_Bcast(
                 /* FIXME binomial may not be the best algorithm for on-node
                    bcast.  We need a more comprehensive system for selecting the
                    right algorithms here. */
-                MPIR_Bcast_fn_or_override(MPIR_Bcast_binomial, mpi_errno_ret,
-                                          buffer, count, datatype,
-                                          MPIR_Get_intranode_rank(comm_ptr, root),
-                                          comm_ptr->node_comm, errflag);
+                mpi_errno = MPID_Bcast(buffer, count, datatype,
+                                       MPIR_Get_intranode_rank(comm_ptr, root),
+                                       comm_ptr->node_comm, errflag);
+                if (mpi_errno) {
+                    /* for communication errors, just record the error but continue */
+                    *errflag = MPIR_ERR_GET_CLASS(mpi_errno);
+                    MPIR_ERR_SET(mpi_errno, *errflag, "**fail");
+                    MPIR_ERR_ADD(mpi_errno_ret, mpi_errno);
+                }
             }
 
             /* perform the internode broadcast */
             if (comm_ptr->node_roots_comm != NULL)
             {
                 if (MPIU_is_pof2(comm_ptr->node_roots_comm->local_size, NULL))
```

Comment:
We can just remove this `if` statement now that both cases do the same thing.

Reply:
Fixed.